Model/ManagerBase.php

```diff
@@ -3,15 +3,46 @@
 namespace Snowdog\CmsApi\Model;
 
 use Magento\Framework\Api\SearchCriteriaInterface;
+use Magento\Framework\UrlInterface;
+use Magento\Store\Model\StoreManagerInterface;
 
 class ManagerBase
 {
+    /**
+     * @var UrlInterface
+     */
+    private $url;
+
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
+     * @param UrlInterface $url
+     * @param StoreManagerInterface $storeManager
+     */
+    public function __construct(
+        UrlInterface $url,
+        StoreManagerInterface $storeManager
+    ) {
+        $this->url = $url;
+        $this->storeManager = $storeManager;
+    }
+
     /**
      * @param SearchCriteriaInterface $searchCriteria
      * @return int
      */
     protected function getStoreIdBySearchCriteria(SearchCriteriaInterface $searchCriteria)
     {
+        $url = $this->url->getCurrentUrl();
+        preg_match('/\/rest\/(.*?)\/V1\//', $url, $scope);
```

Comment:
Use `\Magento\Framework\App\State::getAreaCode()` to check what the area is:

* `\Magento\Framework\App\Area::AREA_WEBAPI_REST`
* `\Magento\Framework\App\Area::AREA_WEBAPI_SOAP`